Fixed rpm versioning for docker

tboerger opened the pull request:
Fixed a typo in the docker experimental versioning format, it should be
~dev instead of ~def.

Signed-off-by: Thomas Boerger thomas@webhippie.de

tboerger committed:
Fixed rpm versioning for docker

Fixed a typo in the docker experimental versioning format, it should be
~dev instead of ~def.

Signed-off-by: Thomas Boerger <thomas@webhippie.de>

cyphar commented:
Can we also fix this in `containerd` and `runc`. I'm not sure we're doing things right there either.

tboerger commented:
> Can we also fix this in containerd and runc. I'm not sure we're doing things right there either.

All version detections are based on `VERSION` files, just docker is a special case where `-dev` gets replaced with `~dev` to make it a valid RPM version.

cyphar commented:
But `runc` has `1.0.0-rc1` in its `VERSION` file (and it used to have `0.1.1-dev`). `containerd` doesn't have a `VERSION` file, to be fair.

jordimassaguerpla commented:
regarding docker, this LGTM . Feel free to open another PR for containerd and runc or to use this one and add some extra commits.

tboerger commented:
> But runc has 1.0.0-rc1 in its VERSION file (and it used to have 0.1.1-dev. containerd doesn't have a VERSION file, to be fair.

Ok, true. That should be fixed in a similar way.

tboerger committed:
Fixed versioning for runc

Since the version of runc includes a -rcX I'm replacing it not also with
~dev.

Signed-off-by: Thomas Boerger <thomas@webhippie.de>

jordimassaguerpla commented:
lgtm

jordimassaguerpla commented:
lgtm
